Sources/HTTPTypes/HTTPRequest.swift

```diff
@@ -224,6 +224,106 @@ extension HTTPRequest: CustomDebugStringConvertible {
     }
 }
 
+extension HTTPRequest.PseudoHeaderFields: Codable {
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.unkeyedContainer()
+        try container.encode(self.method)
+        if let scheme = self.scheme {
+            try container.encode(scheme)
+        }
+        if let authority = self.authority {
+            try container.encode(authority)
+        }
+        if let path = self.path {
+            try container.encode(path)
+        }
+        if let extendedConnectProtocol = self.extendedConnectProtocol {
+            try container.encode(extendedConnectProtocol)
+        }
+    }
+
+    private enum DecodingError: Error {
+        case missingMethod
+        case multipleMethod
+        case multipleScheme
+        case multipleAuthority
+        case multiplePath
+        case multipleExtendedConnectProtocol
+        case invalidMethod(String)
+        case notPseudo(String)
+    }
+
+    public init(from decoder: Decoder) throws {
+        var container = try decoder.unkeyedContainer()
+        var method: HTTPField?
+        var scheme: HTTPField?
+        var authority: HTTPField?
+        var path: HTTPField?
+        var extendedConnectProtocol: HTTPField?
+        while !container.isAtEnd {
+            let field = try container.decode(HTTPField.self)
+            switch field.name {
+            case .method:
+                guard method == nil else {
+                    throw DecodingError.multipleMethod
+                }
+                method = field
+            case .scheme:
+                guard scheme == nil else {
+                    throw DecodingError.multipleScheme
+                }
+                scheme = field
+            case .authority:
+                guard authority == nil else {
+                    throw DecodingError.multipleAuthority
+                }
+                authority = field
+            case .path:
+                guard path == nil else {
+                    throw DecodingError.multiplePath
+                }
+                path = field
+            case .protocol:
+                guard extendedConnectProtocol == nil else {
+                    throw DecodingError.multipleExtendedConnectProtocol
+                }
+                extendedConnectProtocol = field
+            default:
+                guard field.name.rawName.hasPrefix(":") else {
+                    throw DecodingError.notPseudo(field.name.rawName)
+                }
```

Comment:
Do we intentionally discard the fields here?

Reply:
Yes, it's possible to add new pseudo header fields in the future, but it would have no defined meaning in the current library.